overlord/snapstate/autorefresh.go

```diff
@@ -254,54 +256,74 @@ func (m *autoRefresh) Ensure() error {
 		logger.Debugf("Next refresh scheduled for %s.", m.nextRefresh.Format(time.RFC3339))
 	}
 
-	// should we hold back refreshes?
-	holdTime, err := m.EffectiveRefreshHold()
+	held, holdTime, err := m.isRefreshHeld(refreshSchedule)
```

Comment:
Would it make sense to pass `now` or time.Now() as a parameter rather than get it inside?

Reply:
hmm not sure I want to do that, this code already has `time.Now()` spewed every which place so I'm not sure it makes sense to try and clean that up now, otoh it may actually be intentional that every time we want to refer to _now_ semantically we actually go calculate now precisely then. if @pedronis wants this then I will do it but otherwise I'd really like to just land this PR and move on with the rest of the code I have for console-conf things